LanguageFeatures/nnbd/const_evaluation_A10_t01.dart

```diff
@@ -0,0 +1,132 @@
+typedef RecordDynamicNull = (dynamic, {Null n});
+typedef RecordNullableDynamicNullableNever = (
+  NullableDynamic, {
+  NullableNever n
+});
+
+testXExtendsT1<X extends Never>() {
+  Expect.identical(X, Never); // NORM(X extends T) = Never if T is Never
+}
+
+testXExtendsT2<X extends Y, Y extends Never>() {
+  Expect.identical(X,
+      Never); // NORM(X extends T) = Never if T is a type variable Y and NORM(Y) is Never
+}
+
+main() {
+  Expect.identical(
+      Void,
+      FutureOr<void>); // NORM(FutureOr<T>) == S, if NORM(T) == S and S is a top type
+  Expect.identical(
+      dynamic,
+      FutureOr<dynamic>); // NORM(FutureOr<T>) == S, if NORM(T) == S and S is a top type
+  Expect.identical(
+      NullableObject,
+      FutureOr<Object?>); // NORM(FutureOr<T>) == S, if NORM(T) == S and S is a top type
+  Expect.identical(Object,
+      FutureOr<Object>); // NORM(FutureOr<T>) == Object, if NORM(T) == Object
+  Expect.identical(
+      Future<Never>,
+      FutureOr<Never>); // NORM(FutureOr<T>) == Future<Never>, if NORM(T) == Never
+  Expect.identical(NullableFutureNull,
+      FutureOr<Null>); // NORM(FutureOr<T>) == Future<Null>?, if NORM(T) == Null
+  Expect.identical(
+      FutureOr<int?>,
+      FutureOr<DoubleNullableInt>); // NORM(FutureOr<T>) == FutureOr<S>, where NORM(T) == S (all other cases)
+
+  Expect.identical(dynamic,
+      NullableDynamic); // NORM(T?) == S, if NORM(T) == S and S is a top type
+  Expect.identical(
+      Void, NullableVoid); // NORM(T?) == S, if NORM(T) == S and S is a top type
+  Expect.identical(NullableObject,
+      DoubleNullableObject); // NORM(T?) == S, if NORM(T) == S and S is a top type
+  Expect.identical(
+      Null, NullableNever); // NORM(T?) == Null, if NORM(T) == Never
+  Expect.identical(Null, NullableNull); // NORM(T?) == Null, if NORM(T) == Null
+  Expect.identical(FutureOr<int?>,
+      NullableFutureOrNullableInt); // NORM(T?) == FutureOr<R>, if NORM(T) == FutureOr<R> and R is nullable
+  Expect.identical(
+      NullableInt, DoubleNullableInt); // NORM(T?) == R?, if NORM(T) == R?
+  Expect.identical(NullableInt,
+      NullableInt); // NORM(T?) == S?, if NORM(T) == S (all other cases)
+
+  // NORM(C<T0, ..., Tn>) = C<R0, ..., Rn> where Ri is NORM(Ti)
+  Expect.identical(Map<void, dynamic>, Map<FutureOr<void>, FutureOr<dynamic>>);
+  Expect.identical(
+      Map<Object?, Object>, Map<FutureOr<Object?>, FutureOr<Object>>);
+  Expect.identical(
+      Map<Future<Never>, Future<Null>?>, Map<FutureOr<Never>, FutureOr<Null>>);
+  Expect.identical(Map<FutureOr<int?>, dynamic>,
+      Map<FutureOr<DoubleNullableInt>, NullableDynamic>);
+  Expect.identical(Map<void, Object?>, Map<NullableVoid, DoubleNullableObject>);
+  Expect.identical(Map<Null, Null>, Map<NullableNever, NullableNull>);
+  Expect.identical(
+      Map<FutureOr<int?>, int?>, Map<FutureOr<int?>?, DoubleNullableInt>);
+  Expect.identical(Map<F1, F2>, Map<F1_NORM, F2_NORM>);
```

Comment:
Wouldn't the consistent ordering be this one?:
```suggestion
  Expect.identical(Map<F1_NORM, F2_NORM>, Map<F1, F2>);
```